src/Operation/CreateSearchIndexes.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/*
+ * Copyright 2015-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace MongoDB\Operation;
+
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\RuntimeException as DriverRuntimeException;
+use MongoDB\Driver\Server;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Exception\UnsupportedException;
+use MongoDB\Model\SearchIndexInput;
+
+use function array_is_list;
+use function array_map;
+use function current;
+use function is_array;
+use function is_object;
+use function sprintf;
+
+/**
+ * Operation for the createIndexes command.
+ *
+ * @see \MongoDB\Collection::createSearchIndex()
+ * @see \MongoDB\Collection::createSearchIndexes()
+ * @see https://mongodb.com/docs/manual/reference/command/createSearchIndexes/
+ */
+class CreateSearchIndexes implements Executable
+{
+    /** @var string */
+    private $databaseName;
+
+    /** @var string */
+    private $collectionName;
+
+    /** @var array */
+    private $indexes = [];
+
+    /**
+     * Constructs a createSearchIndexes command.
+     *
+     * @param string             $databaseName   Database name
+     * @param string             $collectionName Collection name
+     * @param list<array|object> $indexes        List of search index specifications
+     * @throws InvalidArgumentException for parameter parsing errors
+     */
+    public function __construct(string $databaseName, string $collectionName, array $indexes)
+    {
+        if (! array_is_list($indexes)) {
+            throw new InvalidArgumentException('$indexes is not a list');
+        }
+
+        foreach ($indexes as $i => $index) {
+            if (is_object($index)) {
+                $index = (array) $index;
+            } elseif (! is_array($index)) {
+                throw InvalidArgumentException::invalidType(sprintf('$index[%d]', $i), $index, 'array');
```

Comment:
Note that you can also specify an array as the type argument:
```suggestion
                throw InvalidArgumentException::invalidType(sprintf('$index[%d]', $i), $index, ['object', 'array']);
```

Reply:
The array handling dates back to https://github.com/mongodb/mongo-php-library/commit/b2cd2866c64690603896b788cd56e45ff4cc69a4 and was bundled in with some CSFLE changes for [PHPLIB-511](https://jira.mongodb.org/browse/PHPLIB-511).

I'll defer to you guys if we'd like a ticket to formally remove this. The few times I used it myself was because @alcaeus reminded me it was available in a code review -- but writing list of types as a string myself was never a pain point.